File: hestia/domain/rag/classification.py
from __future__ import annotations

from enum import Enum


class Classification(Enum):
    PUBLIC       = (0, ["public", "public (pu)", "pu", "public (ex)", "ex"])
    INTERNAL     = (1, ["internal", "internal (in)", "in", "interne (in)", "interne"])
    RESTRICTED   = (2, ["restricted", "restricted (re)", "re", "restreint (re)", "restreint"])
    CONFIDENTIAL = (3, ["confidential", "confidential (co)", "co", "confidentiel (co)", "confidentiel"])
    SECRET       = (4, ["secret", "secret (se)", "se"])

    def __init__(self, level: int, aliases: list[str]):
        self.level = level
        self.aliases = [a.lower() for a in aliases]

    @classmethod
    def from_label(cls, label: str) -> Classification | None:
        cleaned = label.strip().lower()
        for item in cls:
            if cleaned in item.aliases:
                return item
        return None

File: hestia-ui/src/lib/classification.ts
// Numeric levels must match hestia/domain/rag/classification.py's Classification
// enum exactly (PUBLIC=0, INTERNAL=1, RESTRICTED=2, CONFIDENTIAL=3, SECRET=4) --
// this is the value stored as a user's clearance / a collection's max_classification
// and compared server-side, so any drift here silently grants/denies the wrong tier
// (see TRP-018).
export const CLASSIFICATION_LABELS: Record<number, string> = {
  0: 'Public',
  1: 'Internal',
  2: 'Restricted',
  3: 'Confidential',
  4: 'Secret',
};

export const CLASSIFICATION_OPTIONS = [0, 1, 2, 3, 4] as const;

export function classificationLabel(level: number | null): string {
  if (level === null) return 'No cap';
  return CLASSIFICATION_LABELS[level] ?? `Level ${level}`;
}

// Per-document classification labels, sent as a metadata string override and
// matched against hestia/domain/rag/classification.py's Classification aliases.
// Kept as string values rather than the numeric levels above: document tagging
// resolves through label matching, independent of the numeric clearance/cap
// assigned to users and collections.
export const DOCUMENT_CLASSIFICATION_OPTIONS: { value: string; label: string }[] = [
  { value: 'public',       label: 'Public' },
  { value: 'internal',     label: 'Internal' },
  { value: 'restricted',   label: 'Restricted' },
  { value: 'confidential', label: 'Confidential' },
  { value: 'secret',       label: 'Secret' },
];
